src/praisonai-code/praisonai_code/cli/features/self_manage.py

```
@@ -0,0 +1,260 @@
+                num += ch
+            else:
+                break
+        parts.append(int(num) if num else 0)
+    return tuple(parts)
+
+
+def is_newer(latest: str, current: str) -> bool:
+    """Return True when ``latest`` is a strictly newer version than ``current``."""
+    try:
+        return _version_tuple(latest) > _version_tuple(current)
+    except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)